### PR TITLE
Add user details in login response

### PR DESCRIPTION
**Pull request by alphinroy-kv**

### CODE REVIEW

**Review by doylefermi-kv — CHANGES_REQUESTED** [review]
Lets's make use of the existing type `User`

**Comment by JoyalAJohney-KeyValue** [comment]
LGTM, @doylefermi-kv any further changes?

File: src/authentication/authentication.graphql [hunk 1]
```diff
@@ -39,6 +39,7 @@ input UserOTPLoginInput {
 type TokenResponse {
   refreshToken: String!
   accessToken: String!
+  user: User
```
Comment (doylefermi-kv) [inline src/authentication/authentication.graphql:42]: Can we ensure that user is mandatory in the schema?

**Review by doylefermi-kv — APPROVED** [review]

File: src/authentication/service/token.service.ts [hunk 2]
```diff
@@ -21,22 +21,20 @@ export class TokenService {
...
-      userRecord,
-    );
+    const token = this.authenticationHelper.generateTokenForUser(userRecord);
     await this.userService.updateField(
       userRecord.id,
       'refreshToken',
       token.refreshToken,
     );
-    return token;
+    return { ...token, user: userRecord };
```
Comment (doylefermi-kv) [inline src/authentication/service/token.service.ts:38]: Can we create a new variable of type TokenResponse and return it? Let's not skip 1 line of code compromising readability.

File: test/authentication/resolver/userauth.resolver.test.ts [hunk 3]
```diff
@@ -81,6 +82,13 @@ describe('Userauth Module', () => {
           username: 'user@test.com',
           password: 's3cr3t1234567890',
         };
+        const user = {
+          id: 'ae032b1b-cc3c-4e44-9197-276ca877a7f8',
```
Comment (doylefermi-kv) [inline test/authentication/resolver/userauth.resolver.test.ts:86]: Let's not reintroduce a new string. Let's reuse the existing `users[]`. The only difference is, this local variable will not have the password and other unwanted details.

Example: [Code reference](https://github.com/KeyValueSoftwareSystems/authentication-service/blob/ea9b3ec6593f92ae1b545c75aed6f32ebbc5dfed/test/authentication/resolver/userauth.resolver.test.ts#L124-L130)

**Review by doylefermi-kv — CHANGES_REQUESTED** [review]

File: src/authentication/service/token.service.ts [hunk 4]
```diff
@@ -21,22 +21,22 @@ export class TokenService {
...
 
   async resetToken(id: string): Promise<void> {
     await this.userService.updateField(id, 'refreshToken', '');
   }
 
   async getNewToken(userRecord: User): Promise<TokenResponse> {
-    const token: TokenResponse = this.authenticationHelper.generateTokenForUser(
-      userRecord,
-    );
+    const token = this.authenticationHelper.generateTokenForUser(userRecord);
```
Comment (doylefermi-kv) [inline src/authentication/service/token.service.ts:33]: Please ignore this line since there is no code change.

**Review by doylefermi-kv — APPROVED** [review]

**Pull request merged**